src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -1166,6 +1166,26 @@ void goto_instrument_parse_optionst::instrument_goto_program()
       goto_model, file_name, ui_message_handler);
   }
 
+  loop_contract_configt loop_contract_config = {
+    cmdline.isset(FLAG_LOOP_CONTRACTS),
+    !cmdline.isset(FLAG_LOOP_CONTRACTS_NO_UNWIND)};
+
+  if(
+    cmdline.isset(FLAG_LOOP_CONTRACTS) &&
+    cmdline.isset(FLAG_LOOP_CONTRACTS_NO_UNWIND))
+  {
+    // When the model is produced by Kani, we must not automatically unwind
+    // the backjump introduced by the loop transformation.
+    // Automatic unwinding duplicates assertions found in the loop body, and
+    // since Kani expects property identifiers to remain unique. Having
+    // duplicate instances of the assertions makes Kani fail to handle the
+    // analysis results.
+    log.warning() << "**** WARNING: transformed loops will not be unwound "
+                  << "after applying loop contracts. Remember to unwind "
+                  << "them at least twice to pass unwinding-assertions."
+                  << messaget::eom;
+  }
```

Comment:
I'm a little concern about this. Why would users need to unwind loop at least twice? The advice should be they must unwind the loops to pass unwinding assertions. Also, this option is not set by default, correct? Why is this Kani specific? Maybe we should re-phrase the documentation to be more general. For instance, I don't see a problem for us to require users to unwind the the remaining loops after loop contract instrumentation, maybe we can even report to them the remaining loops, we don't need to justify this by Kani since it's only a user of the feature, right?

Reply:
After the instrumentation, the transformed will contains a back jump. Running CBMC with unwinding bound 1 will fail the unwinding assertions. Not providing unwinding bounds or providing bounds greater than 1 are fine. 

By default, goto-instrument will unwinding the instrumented loops by 2. 

It is not Kani specific. I will update the documentation and clarify the motivation of this flag.